app.js

```diff
@@ -1,20 +1,28 @@
 const express = require('express');
 const {
     getTopics,
-    getApi
+    getApi,
+    getAllArticles
 } = require('./Controllers/controllers');
 
 const app = express();
 
 
 app.get("/api/topics", getTopics)
 app.get("/api", getApi)
+app.get("/api/articles", getAllArticles)
 
 app.all("/*", (request, response) => {
-    response.status(404).send({message: "Not found"})
+    response.status(404).send({ message: "Not found" })
 })
 
 app.use((err, req, res, next) => {
-    res.status(500).send({message: "Internal server error"});
+    if (err.code === 404) {
+        response.status(404)
+        .send({message: `No matches found in ` + (req.url)})
+    }
```

Comment:
You should add an `else` case here. If the code doesn't match then this will silently swallow the error and it'll never make it to your 500 handler